client/src/main/java/org/apache/rocketmq/client/impl/factory/MQClientInstance.java

```diff
@@ -239,6 +243,8 @@ public void start() throws MQClientException {
                     this.pullMessageService.start();
                     // Start rebalance service
                     this.rebalanceService.start();
+                    //Start heartbeat service
+                    this.heartbeatService.start();
```

Comment:
no shutdown heartbeatService thread when the MQClientInstance shutdown

Reply:
> no shutdown heartbeatService thread when the MQClientInstance shutdown

i will fix it